### DISCUSSION TITLE
How to test a prompt xontrib?

### DISCUSSION THREAD

**Discussion by Spitfire1900** [xonsh/xonsh#5392] [Q&A]
Hello, I am working on a new xontrib, and I'm about ready to release it but I want to write tests first.

My xontrib adds a number of Prompts to the session.

When testing with the xession pytest fixture though I've noticed that only the Prompts that ship out-of-the-box with xonsh are included and not the ones that are installed by 

```python
@pytest.fixture
def prompts(xession: XonshSession):
    #
    print(type(xession.env))
    fields: Env
    fields = xession.env["PROMPT_FIELDS"]  # type: ignore
    yield fields
    fields.clear()

def test_it_loads(load_xontrib):
    load_xontrib("pygitstatus")  # PASSES

def test_clean(prompts, git_repo):
    with cd(git_repo):
        Path('text.txt').touch()

        load_xontrib("pygitstatus") 

        breakpoint() # NO pygitstatus in prompts.
        print('')

```

My xontrib has an entry point so xonsh installed in my projects venv autoloads the PROMPTS normally

```xonsh
$PROMPT_FIELDS
xonsh.prompt.__amalgam__.PromptFields(
 {'user': 'kyle',
  'prompt_end': '$',
  'hostname': 'mainstay',
  'localtime': <function xonsh.prompt.__amalgam__._localtime>,
  'last_return_code': <function xonsh.prompt.__amalgam__.PromptFields.load_initial.<locals>.<lambda>>,
  'last_return_code_if_nonzero': <function xonsh.prompt.__amalgam__.PromptFields.load_initial.<locals>.<lambda>>,
  'gitstatus.short_head': <Prompt: gitstatus.short_head>,
  'gitstatus.tag': <Prompt: gitstatus.tag>,
  'gitstatus.tag_or_hash': <Prompt: gitstatus.tag_or_hash>,
.......
  'pygitstatus': <Prompt: pygitstatus>,
  'pygitstatus_curr_branch': <function xontrib.pygitstatus.prompts.curr_branch>,
  'pygitstatus.ahead': <Prompt: pygitstatus.ahead>,
  'pygitstatus.behind': <Prompt: pygitstatus.behind>,
  'pygitstatus_branch': <Prompt: pygitstatus.branch>
  ....
  'pygitstatus.stash_count': <Prompt: pygitstatus.stash_count>,
  'pygitstatus.tag': <Prompt: pygitstatus.tag>,
  'pygitstatus.tag_or_hash': <Prompt: pygitstatus.tag_or_hash>,
  'pygitstatus.untracked': <Prompt: pygitstatus.untracked>})
```

**Comment by Spitfire1900** [comment]
Anyone have experience writing tests for xontribs that provide prompts?

**Reply by anki-code** [reply]
Hey! If you see no similar xontribs in https://github.com/topics/xontrib so I think the answer is no :)
You can also take a look into xonsh core tests - https://github.com/xonsh/xonsh/tree/main/tests

**Reply by jnoortheen** [reply]
@Spitfire1900 you can take a look here https://github.com/jnoortheen/xontrib-powerline3/blob/main/tests/test_processor.py . Also it seems you are using an older version of Xonsh (with `__amalgam__`). Please update to the latest 0.17.0 and be ready to update to 0.18.0 because it will have refactoring.
